### ISSUE TITLE
os: TestReaddirnamesOneAtATime failed on windows-amd64-2016

### ISSUE THREAD

**Issue by alexbrainman** [golang/go#24742]
TestReaddirnamesOneAtATime failed on windows-amd64-2016 at 33eaf75a031694e11981218963f395174baca6e1

https://build.golang.org/log/03e86c1ca81554f89cb96158bceb0be6f2376f71

```
--- FAIL: TestReaddirnamesOneAtATime (0.04s)
	os_test.go:508: small read "PerfStringBackup.TMP" mismatch: pidgenx.dll
	os_test.go:508: small read "pidgenx.dll" mismatch: PING.EXE
	os_test.go:508: small read "PING.EXE" mismatch: PkgMgr.exe
	os_test.go:508: small read "PkgMgr.exe" mismatch: pku2u.dll
	os_test.go:508: small read "pku2u.dll" mismatch: pl-PL
	os_test.go:508: small read "pl-PL" mismatch: pla.dll
...
...
...
	os_test.go:508: small read "XpsPrint.dll" mismatch: zh-CN
	os_test.go:508: small read "zh-CN" mismatch: zh-HK
	os_test.go:508: small read "zh-HK" mismatch: zh-TW
FAIL
FAIL	os	2.772s
```

Alex

**Comment by ianlancetaylor** [comment]
CC @alexbrainman

**Comment by alexbrainman** [comment]
Sorry @ianlancetaylor I did not have time to investigate issue on the day I reported it.

The TestReaddirnamesOneAtATime requires

// big directory that doesn't change often.

and we have chosen c:\windows\system32 as this is where Windows is installed. But looking at the failure message, it appears that some temp file was written into that directory in the middle of the test. I cannot think anything more suitable but c:\windows\system32 for this test - unless we want to create temp directory with many files specifically for that test.

This test is fairly old, but I do not remember seen this problem before. So I hope it will not happen soon enough to bother us again.

Closing this as unfortunate fake.

Alex
